library/blob.c

```diff
@@ -491,6 +695,23 @@ static int eblob_check_disk_one(struct eblob_iterate_local *loc)
 		}
 	}
 
+	if ((ctl->flags & EBLOB_ITERATE_FLAGS_VERIFY_CHECKSUM) &&
+	    !(dc->flags & BLOB_DISK_CTL_REMOVE) &&
+	    !(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+		struct eblob_write_control wc;
+		memset(&wc, 0, sizeof(wc));
+		eblob_rctl_to_wc(&rc, &wc);
+		eblob_dc_to_wc(dc, &wc);
+
+		err = eblob_verify_checksum(bc->back, &dc->key, &wc);
+		if (err) {
+			eblob_dump_wc(bc->back, &dc->key, &wc, "eblob_check_disk_one: checksum verification failed", err);
+			eblob_mark_entry_removed_purge(bc->back, &dc->key, &rc);
```

Comment:
Here is the discussion I refer to: https://github.com/reverbrain/eblob/pull/157

I only have following recovery options so far:
* remove key from index/data if headers are mismatched in index/data headers, for example when keys are different, when sizes are outside of valid values (like more than size of the blob)
* added no-checksum bit if there is no footer or its size is not big enough to have checksum in it
* fixing index/data sizes if they mismatch, new size should correspond to what we have in data (this is questionable enough - one could update only index while overwriting data, but afaics this can not happen, so data size is always valid)
* clearing ext-header bit if there is not enough space
* make both index and data header the same, for example recover key (clear 'removed' bit) if key is present in data and is removed in index. this can be reversed though (removing data if it is removed in index), main purpose it to make both index and data header for given key the same

What options do you think we should add?